### PR TITLE
Add Multi Window and Titlebar samples

### PR DESCRIPTION
**Pull request by bpulliam**
## Description
This change improves XamlControlsGallery's Multi-Window support.

Update XCG to use WinAppSDK 1.0.1 and CsWinRT 1.5.0.

NavigationRootPage now passes a reference to itself to the child page when it navigates.  It also replaces the Current attribute with GetForElement() so that NavigationRootPage is no longer assumed to be a singleton. Much of the change is moving to this new system.

Uses of App.CurrentWindow have been replaced with Window.Current.

A new class named WindowHelper tracks the Windows on the thread so that given a random UIElement, we can find the Window it lives in.  CreateWindow is also present here so that the helper can maintain that list.

A new class has been added to provide interop with win32 calls:
- GetActiveWindow
- GotModuleHandle
- LoadIcon
- SendMessage

ThemeHelper has been updated to work with multiple windows.

IdleSynchronizer has been updated to work with multiple windows.

The NavigationOrientationHelper has changed IsLeftMode from an attribute to a method.

Add a new example to show the creation of multiple windows.

Add a new example to show the various customization options for the titlebar (system, custom, fallback).

Allow the TabViewPage to create a new Window again

## Motivation and Context
WinAppSDK 1.0.1 fixed multiple issues that prevented proper use of multiple windows and  titlebars.  The fixing of these issues has unblocked the use of both of these in XCG.

## How Has This Been Tested?
Manual testing

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: XamlControlsGallery/Common/Win32.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+        [DllImport("user32.dll")]
+        public static extern IntPtr GetActiveWindow();
+
+        [DllImport("kernel32.dll", CharSet = CharSet.Auto)]
+        public static extern IntPtr GetModuleHandle(IntPtr moduleName);
+
+        public const int WM_ACTIVATE = 0x0006;
+        public const int WA_ACTIVE = 0x01;
+        //static int WA_CLICKACTIVE = 0x02;
```
Comment (kmahone) [inline XamlControlsGallery/Common/Win32.cs:22]: Why is this commented out?

File: XamlControlsGallery/ControlPages/AcrylicPage.xaml.cs [hunk 2]
```diff
@@ -27,7 +27,7 @@ private void Slider_ValueChanged(object sender, RangeBaseValueChangedEventArgs e
         {
             //Rectangle shape = (sender == OpacitySliderInApp) ? CustomAcrylicShapeInApp : CustomAcrylicShape;
             Rectangle shape = CustomAcrylicShapeInApp;
-            if (sender == OpacitySliderLumin)
+            if ((Slider)sender == OpacitySliderLumin)
```
Comment (kmahone) [inline XamlControlsGallery/ControlPages/AcrylicPage.xaml.cs:30]: Why is this change necessary? I would expect comparing 'object' to 'Slider' should work without the need to down cast.

File: XamlControlsGallery/Navigation/NavigationRootPage.xaml.cs [hunk 3]
```diff
@@ -36,17 +36,26 @@ namespace AppUIBasics
...
         public Windows.System.VirtualKey ArrowKey;
+        public Frame RootFrame { get { return rootFrame; }}
 
         private RootFrameNavigationHelper _navHelper;
         private bool _isGamePadConnected;
         private bool _isKeyboardConnected;
         private Microsoft.UI.Xaml.Controls.NavigationViewItem _allControlsMenuItem;
         private Microsoft.UI.Xaml.Controls.NavigationViewItem _newControlsMenuItem;
 
+        public static NavigationRootPage GetForElement(object obj)
```
Comment (kmahone) [inline XamlControlsGallery/Navigation/NavigationRootPage.xaml.cs:48]: >object [](http://example.com/codeflow?start=55&length=6)

Should this be typed as UIElement?

File: XamlControlsGallery/ControlPages/CustomTitlebarPage.xaml [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+            <local:ControlExample.Example>
+                <StackPanel Orientation="Vertical" Spacing="10">
+                    <TextBlock TextWrapping="WrapWholeWords">
+                        User can set a top-level UIElement (defined as appTitlebar here) as titlebar for the window. The system titlebar disappears and the chosen uielement starts acting like the titlebar. <LineBreak></LineBreak>
+                        The Background and Foreground Color dropdowns set the foreground and background of titlebar and caption buttons respectively.
+                    </TextBlock>
+                        <StackPanel Orientation="Horizontal" HorizontalAlignment="Stretch" VerticalAlignment="Top" Spacing="10">
+                            <Button x:Name="customTitlebar" Click="customTitlebar_Click"></Button>
+                        <TextBlock> Background Color </TextBlock>
+                        <SplitButton x:Name="myBgColorButton" AutomationProperties.Name="Foreground color" Padding="0" MinHeight="0" MinWidth="0" VerticalAlignment="Top">
```
Comment (kmahone) [inline XamlControlsGallery/ControlPages/CustomTitlebarPage.xaml:40]: Automation Name for background and foreground buttons appear to be switched.

File: XamlControlsGallery/ControlPages/CustomTitlebarPage.xaml.cs [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+
+        private void UpdateTitlebarColor()
+        {
+            var res = Microsoft.UI.Xaml.Application.Current.Resources;
+            res["WindowCaptionBackground"] = currentBgColor;
+            //res["WindowCaptionBackgroundDisabled"] = currentBgColor;
+            res["WindowCaptionForeground"] = currentFgColor;
+            //res["WindowCaptionForegroundDisabled"] = currentFgColor;
+
+            // to trigger repaint tracking task id 38044406
```
Comment (kmahone) [inline XamlControlsGallery/ControlPages/CustomTitlebarPage.xaml.cs:129]: I assume that this is a bug that we are working around. We should add a comment explaining what the bug is. XCG is used as documentation and readers will likely copy any code that is included here.

File: XamlControlsGallery/ControlPages/CustomTitlebarPage.xaml.cs [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+        private void BgColorButton_Click(object sender, RoutedEventArgs e)
+        {
+            // Extract the color of the button that was clicked.
+            Button clickedColor = (Button)sender;
+            var rectangle = (Microsoft.UI.Xaml.Shapes.Rectangle)clickedColor.Content;
+            var color = ((Microsoft.UI.Xaml.Media.SolidColorBrush)rectangle.Fill).Color;
+
+            CurrentBgColor.Background = new SolidColorBrush(color);
+
+            currentBgColor = color;
```
Comment (kmahone) [inline XamlControlsGallery/ControlPages/CustomTitlebarPage.xaml.cs:101]: The variable naming here confused me for a moment, since we have 'currentBgColor' and 'CurrentBgColor' that differ only in case, and 'CurrentBgColor' is not a Color, it is a Border. I suggest updated 'CurrentBgColor' to something like 'BackgroundColorElement'.

Similarly with CurrentFgColor/currentFgColor.

File: XamlControlsGallery/ControlPages/CreateMultipleWindowsPage.cs [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Navigation;
+
+// The Blank Page item template is documented at http://go.microsoft.com/fwlink/?LinkId=234238
+
+namespace AppUIBasics.ControlPages
+{
+    /// <summary>
+    /// An empty page that can be used on its own or navigated to within a Frame.
+    /// </summary>
```
Comment (kmahone) [inline XamlControlsGallery/ControlPages/CreateMultipleWindowsPage.cs:23]: Remove comment.

File: XamlControlsGallery/ControlPages/CreateMultipleWindowsPage.cs [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+//*********************************************************
+using AppUIBasics.Data;
+using AppUIBasics.Helper;
+using System.Collections.Generic;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Navigation;
+
+// The Blank Page item template is documented at http://go.microsoft.com/fwlink/?LinkId=234238
+
```
Comment (kmahone) [inline XamlControlsGallery/ControlPages/CreateMultipleWindowsPage.cs:18]: Remove comment.

File: XamlControlsGallery/ControlPages/CreateMultipleWindowsPage.cs [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+    /// </summary>
+    public sealed partial class CreateMultipleWindowsPage : Page
+    {
+        private IEnumerable<ControlInfoDataGroup> _groups;
+
+        public CreateMultipleWindowsPage()
+        {
+            this.InitializeComponent();
+        }
+        public IEnumerable<ControlInfoDataGroup> Groups
```
Comment (kmahone) [inline XamlControlsGallery/ControlPages/CreateMultipleWindowsPage.cs:32]: Consider adding a comment explaining what this is for and who is consuming it.

File: XamlControlsGallery/ControlPages/CustomTitlebarPage.xaml.cs [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+            UpdateTitlebarColor();
+
+        }
+
+
+        private void UpdateTitlebarColor()
+        {
+            var res = Microsoft.UI.Xaml.Application.Current.Resources;
+            res["WindowCaptionBackground"] = currentBgColor;
+            //res["WindowCaptionBackgroundDisabled"] = currentBgColor;
```
Comment (kmahone) [inline XamlControlsGallery/ControlPages/CustomTitlebarPage.xaml.cs:125]: Why are these two lines commented out?

**Review by kmahone — CHANGES_REQUESTED** [review]
:clock1:

File: XamlControlsGallery/Navigation/NavigationRootPage.xaml.cs (hunk 3 above)
Comment (bpulliam) [reply XamlControlsGallery/Navigation/NavigationRootPage.xaml.cs:48]: No, since it supports Application objects as well.  (App.xaml.cs:253)

Comment (kmahone) [reply XamlControlsGallery/Navigation/NavigationRootPage.xaml.cs:48]: But this will fail immediately if passed anything other than a UIElement, since the very first line does a cast.

File: XamlControlsGallery/ControlPages/CustomTitlebarPage.xaml (hunk 4 above)
Comment (bpulliam) [reply XamlControlsGallery/ControlPages/CustomTitlebarPage.xaml]: fixed

File: XamlControlsGallery/Common/Win32.cs (hunk 1 above)
Comment (bpulliam) [reply XamlControlsGallery/Common/Win32.cs:22]: Please resolve this with Pratik

File: XamlControlsGallery/ControlPages/CreateMultipleWindowsPage.cs (hunk 9 above)
Comment (bpulliam) [reply XamlControlsGallery/ControlPages/CreateMultipleWindowsPage.cs]: Removed

File: XamlControlsGallery/ControlPages/CustomTitlebarPage.xaml.cs (hunk 10 above)
Comment (bpulliam) [reply XamlControlsGallery/ControlPages/CustomTitlebarPage.xaml.cs:125]: Please resolve this with Pratik
